Added an option to Settings.

Palingenesis opened the pull request:
### Added an option to Settings.
The option is a Check Box called "Click on select"
The choice only applies during BOM creation, so if the setting is changed a new BOM needs to be created.  
What it does is change the way you interact with the page. If the setting is selected you have to click items to select instead of moving the mouse over.  
The reason for it: "I was sick of knocking my mouse and seeing the wrong items highlighted"

Hopefully I have done things correctly. do check it, I have placed comments where I changed things in the code.
I am OK with html and JS, but don't now how this becomes an addon.

qu1ck commented:
This is impossible to review because of all the formatting changes. Undo irrelevant changes and also remove the `.vs` folder from your git (you can add it to gitignore).

Palingenesis commented:
Sorry I knew I would cock it up.  
It was basically a change at "ibom.js" from line 681.  
```
		var handler = createRowHighlightHandler(tr.id, references, netname);
		/*
			Tim (Palingenesis)
			For choice of using select instead of mouse over
		*/
		//tr.onmousemove = handler;   /*  Tim (Palingenesis)	*/
		if (settings.selectOnClick) {
			tr.onmousedown = handler;
		} else {
			tr.onmousemove = handler;
		}
		/*	End	*/

```
  
In "ibom.html" I added a new Check Box "checkbox_SelectOnClick" after dragCheckbox from 119
```
	<label class="menu-label">
		<input id="dragCheckbox" type="checkbox" checked onchange="setRedrawOnDrag(this.checked)">
		Continuous redraw on drag
	</label>
	<label class="menu-label">
		<input id="checkbox_SelectOnClick" type="checkbox" checked onchange="setSelectOnClick(this.checked)">
		Select On Click<br>(Effects On Create Only)
	</label>
```  
  
I changed/added new setting in the relative files to do with settings to create a new setting called "settings.selectOnClick"
in "util.js" and  "render.js"
  
Can you see what I was doing?
I was adding a choice to select items instead of just moving the mouse over.
Having to keep the mouse over items while you work on the parts is I little silly.
  
Should I try push again?
